Lib/fontbakery/profiles/googlefonts.py

```diff
@@ -4222,6 +4223,42 @@ def com_google_fonts_check_repo_fb_report(family_directory):
                   " since they are ephemeral; FB has a 'github markdown' output mode"
                   " to make it easy to file reports as issues.")
 
+
+@check(
+  id = 'com.google.fonts/check/repo/zip_files',
+  conditions = ['family_directory'],
+  rationale="""
+    Sometimes people check in ZIPs into their font project repositories. While we accept the practice of checking in binaries, we believe that a ZIP is a step too far ;)
+
+    Note: a source purist position is that only source files and build scripts should be checked in. 
+  """,
+  misc_metadata = {
+    'request': 'https://github.com/googlefonts/fontbakery/issues/2903'
+  }
+)
+def com_google_fonts_check_repo_zip_files(family_directory):
```

Comment:
IMHO "upstream repo" is not the same as the here used `family_directory` condition. https://github.com/googlefonts/fontbakery/blob/7e5988af0e152026c79a4779b7d745d23177d2ac/Lib/fontbakery/profiles/shared_conditions.py#L39-L46

It uses `os.path.dirname(font)`, which is usually expected to be the directory in `google/fonts/{license_dir}`.

We should rather try to establish a clear distinction if we want to check within google/fonts, which I consider the default for this profile, or anywhere else, which could be a full new profile e.g. `googlefonts_upstream.py`. Otherwise our understanding of the environment in which we run this Font Bakery Profile is getting more and more fuzzy and nobody will eventually be able to explain what it actually expects.
For the `google/fonts family_directory` I'd rather suggest a whitelist based check, as blacklisting is terribly inefficient and we already have a good understanding and documentation of which files we expect.

I'm not even sure how much "bossing around" in other peoples repos we should do, this is like invading private space. As a best practice hint, it's probably rather a WARN.